### PR TITLE
System time implementation

### PR DESCRIPTION
**Pull request by museun**
This is discussed in #3

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -50,6 +73,16 @@ mod reference {
     pub fn get_time() -> Duration {
         *TIME.get_or_init(Mutex::default).lock().unwrap()
     }
+
+    pub fn get_time() -> Duration {
```
Comment (bryanlarsen) [inline src/lib.rs:77]: there are two get_time functions here.   I assume one will alias the other.

File: src/lib.rs [hunk 2]
```diff
@@ -80,25 +122,135 @@ impl std::fmt::Debug for MockClock {
...
+
+/// A simple deterministic SystemTime wrapped around a modifiable Duration
+///
+/// This used a thread-local state as the 'wall clock' that is configurable via
+/// the `MockClock`
+#[derive(Debug, Copy, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct SystemTime(Duration);
+
+impl SystemTime {
+    pub const UNIX_EPOCH: SystemTime = SystemTime(Duration::from_secs(0));
```
Comment (bryanlarsen) [inline src/lib.rs:193]: Being inside the struct prevents me from accessing it via `use` the way I access std::time::UNIX_EPOCH;

I worked around it with

```
#[cfg(test)]
const UNIX_EPOCH: SystemTime = SystemTime::UNIX_EPOCH;
```

Comment (museun) [reply src/lib.rs:193]: Oh, one should be with_system_time. copy-pasted error (basically get-nnn is a read-only accessor, and with-nnn are a way to modify the value atomically)

Comment (museun) [reply src/lib.rs]: > Being inside the struct prevents me from accessing it via `use` the way I access std::time::UNIX_EPOCH;
> 
> I worked around it with
> 
> ```
> #[cfg(test)]
> const UNIX_EPOCH: SystemTime = SystemTime::UNIX_EPOCH;
> ```

Interesting, I didn't know UNIX_EPOCH was at the module level. std has the struct-level one aliased from the module level one. I added this in 07c1ab2172457672d5ff653625607f486628d649

**Comment by museun** [comment]
If everything looks good now, I'll merge and publish it.

**Pull request merged**